server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -383,44 +392,49 @@ public void parseKnnVectorAndIndex(DocumentParserContext context, DenseVectorFie
                 }
                 fieldMapper.checkDimensionMatches(index, context);
                 checkVectorMagnitude(fieldMapper.fieldType().similarity, errorByteElementsAppender(vector), squaredMagnitude);
+                return VectorData.fromBytes(vector);
+            }
+
+            private VectorData parseHexEncodedVector(DocumentParserContext context, DenseVectorFieldMapper fieldMapper) throws IOException {
```

Comment:
While we support hex-encoded vectors for float fields on query time,  this is currently available only for `byte[]` vectors during indexing. Should we update this as well to also support hex-encoded vectors for float fields during indexing? Should we be worry of incompatible indices in this case?  @benwtrent wdyt?

Reply:
I see what you mean - will leave as is then for indexing. Thanks for the comment @benwtrent !

Reply:
I found it is strange that we for float types, we will support hex-encoded queries but hex-encoded docs.  

I am not sure I understand the reasoning for BWC, would not the same apply for hex-encoded byte vectors: a node < 8.14 can't pass them?

Reply:
> would not the same apply for hex-encoded byte vectors: a node < 8.14 can't parse them?

@mayya-sharipova  

The following are the scenario is what I had in mind.

1. hex query gets parsed on a new node
2. We serialize to an old node, thus having to transform into `float[]`, we have no way of knowing the mapping or anything, but to serialize the query to an old node, we have to transform to `float[]`
3. We are now at the data node with `float[]` in hand & the index is a floating point dense_vector field. There is no way to know if we should throw or not.

Now, we COULD then throw in this scenario once the old node is updated. So, previously, during a mixed cluster environment we allowed an action, but then once its fully upgraded, we will throw?

Seems weird to me.

The answers to me seem like this, either:

A. we allow BWC for hex encoded queries in a mixed cluster environment & allow them against floating point dense_vectors.
B. We never allow BWC for hex encoded queries, requiring that all nodes are upgraded before users can use them.


I voted for `A` as it seems reasonable to me.

Reply:
@benwtrent Thanks, Ben, for a great explanation. It makes sense.  I am less worried about BWC at it is supposed to be short-lived, but because CCS needs to support the previous minor version, it is great to have the option A. 

+1 for that.